server/routes/bookers/booker/index.ts

```diff
@@ -34,23 +36,43 @@ export default function routes(services: Services): Router {
     return BOOKER_REFERENCE_REGEX.test(reference) ? next() : res.redirect('/bookers')
   })
 
-  get('/:reference/', bookerDetails.view())
-  post('/:reference/clear', bookerDetails.clear())
+  // middleware to validate prisoner ID on all /bookers/booker/{:reference}/prisoner/{:prisonerId} routes
```

Comment:
To handle multiple prisoners per booker, I've added prisoner to the URL path, so we end up with `/bookers/booker/{:reference}/prisoner/{:prisonerId}/...`. This middleware validates `prisonerId` for all relevant routes.